File: Data Structures & Algorithms/last-stone-weight/submission-2.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> b - a);
        for (int i = 0; i < stones.length; i++) {
            pq.offer(stones[i]);
        }
        while (pq.size() > 1) {
            int first = pq.poll();
            int second = pq.poll();
            if (first != second) pq.offer(first - second);
        }
        if (pq.isEmpty()) return 0;
        return pq.peek();
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        arr = []
        for char in tokens:
            if char == '+':
                first = arr.pop()
                second = arr.pop()
                arr.append(int(first) + int(second))
            elif char == '-':
                first = arr.pop()
                second = arr.pop()
                arr.append(int(second) - int(first))
            elif char == '*':
                first = arr.pop()
                second = arr.pop()
                arr.append(int(first) * int(second))
            elif char == '/':
                first = arr.pop()
                second = arr.pop()
                arr.append(int(int(second) / int(first)))
            else:
                arr.append(int(char))
        return arr.pop()

        